### ISSUE TITLE
Handle UIControlState better

### ISSUE THREAD

**Issue by ZevEisenberg** [Raizlabs/BonMot#214]
In AdaptableTextContainer.swift:

```swift
extension UIControlState {

    @nonobjc static var allStates: [UIControlState] {
        #if swift(>=3.0)
            return [.normal, .highlighted, .disabled]
        #else
            return [.Normal, .Highlighted, .Disabled]
        #endif
    }

}
```

This excludes `.selected`, which @KingOfBrian thinks was causing runtime errors, but it seems like an egregious omission, so we should look into it. But worse, `UIControlState` is an `OptionSet`, so this approach ignores cases like setting a button's appearance for `[.selected, .highlighted]`. We should investigate ways to more completely represent all of a control's per-state appearance customizations.

**Comment by ZevEisenberg** [comment]
Here's what I'm thinking, after discussing this with @robcadwallader: I'll make a new `StyleableTextControl` protocol for buttons and other controls that typically have `setAttributedTitle(_: NSAttributedString?, for: UIControlState)` and `attributedTitle(for: UIControlState) -> NSAttributedString?` methods. When you call this protocol's methods, which will have a `state` parameter, it will use associated objects to look up UIControlState values in a dictionary like this:

```swift
extension UIControlState: Hashable {

    public var hashValue: Int {
        return Int(rawValue)
    }

}

var dict = [UIControlState: AttributedStringStyle]()
```

**Comment by KingOfBrian** [comment]
closed by #210
